numba/core/types/misc.py

```diff
@@ -334,6 +334,10 @@ def __init__(self, value):
         members = 2 if value.step is None else 3
         SliceType.__init__(self, name=name, members=members)
 
+    @property
+    def key(self):
+        return self.name
```

Comment:
```suggestion
    def key(self):
        sl = self.literal_value
        return sl.start, sl.stop, sl.step
```
using a tuple of actual values perhaps seems more consistent with other keys?

Reply:
Not sure, but it does seem better to return the identity-defining objects themselves instead of their string form.